go/pkg/reader/reader.go

```diff
@@ -187,13 +193,12 @@ func NewCompressedSeeker(fs ReadSeeker) (ReadSeeker, error) {
 		fs:    fs,
 		encdW: encdW,
 		buf:   sb,
-		err:   nil,
 	}, nil
 }
 
 func (cfs *compressedSeeker) Read(p []byte) (int, error) {
-	if cfs.err != nil {
-		return 0, cfs.err
+	if !cfs.IsInitialized() {
+		return 0, errNotInitialized
 	}
 
 	var n int
```

Comment:
Looking at the code block that follows, can we simplify the error handling using [`io.CopyBuffer`](https://pkg.go.dev/io#CopyBuffer)? unlike the current code, `io.CopyBuffer` (and `io.Copy`) internally handles `io.EOF`, so the code below gets simpler as well. I think that means you can write:

```
var err error
if cfs.encdW != nil && cfs.buf.Len() < len(p) { // <-- `for` became `if` because `io.CopyBuffer` loops internally.
  _, err1 := io.CopyBuffer(cfs.encdW.Encoder, cfs.fs)
  err2 := cfs.encdW.Encoder.Close()
  encoders.Put(cfs.encdW)
  cfs.encdW = nil
  if err1 != nil {
    err = err1
  } else if err2 != nil {
    err = err2
  }
}
n, err3 := cfs.buf.Read(p)
if err == nil {
  err = err3
}
return n, err
```

Reply:
Oops, that should be `io.CopyBuffer(cfs.encdW.Encoder, cfs.fs, p)`.